src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.CreateToken.cs

```diff
@@ -1335,60 +1335,62 @@ internal IEnumerable<SecurityKey> GetContentEncryptionKeys(JsonWebToken jwtToken
             // keep track of exceptions thrown, keys that were tried
             StringBuilder exceptionStrings = null;
             StringBuilder keysAttempted = null;
-            foreach (var key in keys)
+            if (keys != null)
             {
-                try
+                foreach (var key in keys)
```

Comment:
wondering if we can extract some of this to a helper method, which we can isolate with testing? Maybe is outside of this PR though.

Reply:
We have tests for this `GetContentEncryptionKeys` already. I think it's fairly easy and straightforward to hit all code paths with the current code.